src/Albany_DataTypes.hpp

```diff
@@ -23,12 +24,11 @@
 #include "Sacado_CacheFad_DFad.hpp"
 #include "Phalanx_KokkosDeviceTypes.hpp"
 
-//amb Need to move to configuration.
-//#define ALBANY_SFAD_SIZE 27
-//#define ALBANY_SLFAD_SIZE 27
+#include "TpetraCore_config.hpp"
 
-//#define ALBANY_FAST_FELIX
-// Typedef AD types to standard names
+#ifndef HAVE_TPETRA_INST_DOUBLE
```

Comment:
I would use HAVE_TPETRA_DOUBLE, in case someone does not use (?) ETI in Tpetra.

Reply:
Also for all the int types below.

Reply:
`HAVE_TPETRA_INST_DOUBLE` is actually independent of ETI.  Tpetra "enables" a set of Scalar types, regardless of whether ETI is enabled.  It is incorrect to use Scalar (GlobalOrdinal, etc.) types as template parameters if they have not been enabled.

Reply:
So what's the difference between HAVE_TPETRA_DOUBLE and HAVE_TPETRA_INST_DOUBLE?

Reply:
@bartgol None :-)

Reply:
Ah, ok. Good to know. Any plan to remove one of the two and avoid the confusion in the future? :-)

Reply:
@bartgol Just saw your question now for some reason.  The answer is "possibly yes."  Note that "instantiation" does not necessary imply "explicit instantiation," though :-) .